### PR TITLE
add --dry-run flag

### PR DESCRIPTION
**Pull request by maxenglander**
Add `--dry-run` flag to build script. It does everything except make remote changes to GitHub.

### CODE REVIEW

File: build-vitess-packages.sh [hunk 1]
```diff
@@ -92,44 +104,49 @@ PREFIX=${PREFIX:-/usr}
...
+  -C "${RELEASE_DIR}" \
+  --before-install "$VTROOT/tools/preinstall.sh" \
+  --package "$(dirname "${RELEASE_DIR}")" \
+  --iteration "${SHORT_REV}" \
+  -t rpm
 
 
 echo "Now updating vitess-release-roster.md ...."
 cd /workspaces/vitess-releases
 echo "| $(date +%x) | @${GITHUB_USER} | [${SHORT_REV}](https://github.com/planetscale/vitess-releases/releases/tag/${SHORT_REV}) |" >> vitess-release-roster.md
```
Comment (FancyFane) [inline build-vitess-packages.sh:141]: We may want to add this line below the DRY_RUN if statement below. This way the file isn't modified on a dry run; and no git resets will be needed.

**Review by FancyFane — COMMENTED** [review]
There's one line of code we may want to relocate into an if statement here to make our lives easier.

File: build-vitess-packages.sh (hunk 1 above)
Comment (maxenglander) [reply build-vitess-packages.sh]: Good idea, done!

Comment (FancyFane) [reply build-vitess-packages.sh]: LGTM

**Review by FancyFane — APPROVED** [review]
Overall LGTM

**Pull request merged**